src/bitwise.c

```diff
@@ -39,7 +39,7 @@ static const unsigned int mask8B[]=
 void oggpack_writeinit(oggpack_buffer *b){
   memset(b,0,sizeof(*b));
   b->ptr=b->buffer=_ogg_malloc(BUFFER_INCREMENT);
-  b->buffer[0]='\0';
```

Comment:
This do not look correct.   If malloc() fail, I doubt b->storage should get a non-zero value as the there is no storage available.

Reply:
> This do not look correct. If malloc() fail, I doubt b->storage should get a non-zero value as the there is no storage available.

The fix do the check for a null. You can see this under the review conversation.

Reply:
I do not understand your comment.  I see the addition of `if(b->buffer) ` in your patch, but fail to see how this properly handle the case where malloc() return NULL, as it seem incomplete and returning a misleading state from the oggpack_writeinit() method.  Why is your edition still modifying b->storage if there is no memory area pointed to by b->buffer?

Reply:
I created https://gitlab.xiph.org/xiph/ogg/-/merge_requests/26 with my proposed fix for the same issue, including updating the documentation to mention that the method can fail.

Reply:
> I do not understand your comment. I see the addition of `if(b->buffer) ` in your patch, but fail to see how this properly handle the case where malloc() return NULL

Conditional operator `if (b->buffer)` prevents from writing `'\0'` into `b->buffer` when it is null.

> but fail to see how this properly handle the case where malloc() return NULL, as it seem incomplete and returning a misleading state from the oggpack_writeinit() method.

The commit do exactly what it says - nothing more, nothing less.
I could add another commit to address this exact issue (the incomplete state).

> Why is your edition still modifying b->storage if there is no memory area pointed to by b->buffer?

That was in original code.